cdc/entry/schema_test_helper.go

```diff
@@ -166,13 +166,19 @@ func (s *SchemaTestHelper) DDL2Jobs(ddl string, jobCnt int) []*timodel.Job {
 // DML2Event execute the dml and return the corresponding row changed event.
 // caution: it does not support `delete` since the key value cannot be found
 // after the query executed.
-func (s *SchemaTestHelper) DML2Event(dml string, schema, table string) *model.RowChangedEvent {
+func (s *SchemaTestHelper) DML2Event(dml string, schema, table string, partitionID ...string) *model.RowChangedEvent {
 	s.tk.MustExec(dml)
 
 	tableInfo, ok := s.schemaStorage.GetLastSnapshot().TableByName(schema, table)
 	require.True(s.t, ok)
 
-	key, value := s.getLastKeyValue(tableInfo.ID)
+	tableID := tableInfo.ID
+	partitionTableID := tableInfo.TableInfo.GetPartitionInfo().GetPartitionIDByName(partitionID[0])
```

Comment:
It seems that only `partitionID[0]` will be used.

Reply:
yes, the caller pass in the partition.